src/user/userRepository.ts

```diff
@@ -0,0 +1,48 @@
+import {UserFactory} from 'src/entities/userFactory';
+import {UserData} from 'src/entities/user.interface';
+import {Injectable, Inject} from '@nestjs/common';
+import {SearchUsersDto} from './dto/searchUsers.dto';
+
+@Injectable()
+export class UserRepository {
+    private userFactory: UserFactory;
+
+    constructor(/* {dbProvider, userFactory, cacheProvider} */ @Inject(UserFactory) userFactory) {
+        // i don't get these injects, only after i added @Inject(UserFactory it worked but if i
```

Comment:
Dependency injection is the most powerful feature of nest itself (almost). That's why it requires to be discussed separately. We'll schedule call on this topic, there is a lot of info to write it down here.